src/performance/dashboard/src/components/modals/__tests__/ModalRunTest.test.js

```diff
@@ -0,0 +1,131 @@
+const componentProps = {
+    handleRunTestDlgClose: jest.fn(),
+    handleRunTestDlgStart: jest.fn()
+}
+
+// Initialise redux stores
+const store = createStore(reducers, {
+    projectInfoReducer: { config: DataProjectInfo.projectInfo, receivedAt: Date.now(), fetched: true }
+});
+
+// component to render
+const wrapper = (
+    <Provider store={store}>
+        <ModalRunTest {...componentProps} />
+    </Provider>
+)
+
+// dont leak state
+afterEach(cleanup);
+
+// constants
+const TEST_DESCRIPTION = "A short description describing the test"
+const MAX_DESCRIPTION_LENGTH = 80;
+
+/** 
+ * Test functionality of the ModelRunTest dialog
+ */
+describe('<ModalRunTest />', () => {
+    test('dialog displays without error', () => {
+        render(wrapper)
+        expect(document.querySelector('#ModalRunTest.bx--modal').id).toBe('ModalRunTest');
+    });
+
+    test('modal has a valid heading', () => {
+        render(wrapper)
+        expect(document.querySelector('#ModalRunTest .bx--modal-header__heading').innerHTML).toBe('Load test configuration');
+    });
+
+    test('there is an active cancel button ', () => {
+        const { getByText } = render(wrapper);
+        const cancelButton = getByText('Cancel');
+        expect(cancelButton.disabled).toEqual(false)
+    })
+
+    test('there is an active submit button ', () => {
+        const { getByText } = render(wrapper);
+        const submitButton = getByText('Run')
+        expect(submitButton.disabled).toEqual(false)
+    })
+
+    test('there is an input textarea field', () => {
+        render(wrapper)
+        expect(document.querySelector('#ModalRunTest textarea').placeholder).toBe('Describe this new test');
+    });
+
+    test('the textarea is empty by default', () => {
+        render(wrapper)
+        expect(document.querySelector('.charCounter').innerHTML).toBe(`0 / ${MAX_DESCRIPTION_LENGTH}`);
+    });
+
+    test('the text field can receive input and updates character count', () => {
+        const { getByPlaceholderText } = render(wrapper)
+        const textAreaField = getByPlaceholderText(/Describe this new test/i);
+        fireEvent.input(textAreaField, { target: { value: TEST_DESCRIPTION } });
+        expect(textAreaField.value).toEqual(TEST_DESCRIPTION);
+        expect(document.querySelector('.charCounter').innerHTML).toBe(`${TEST_DESCRIPTION.length} / ${MAX_DESCRIPTION_LENGTH}`);
+    });
+
+    test(`counter warning appears after ${MAX_DESCRIPTION_LENGTH-9} characters typed`, () => {
+        const TEST_DESC =  "a".repeat(MAX_DESCRIPTION_LENGTH - 9);
+        const { getByPlaceholderText } = render(wrapper)
+        const textAreaField = getByPlaceholderText(/Describe this new test/i);
+        fireEvent.input(textAreaField, { target: { value: TEST_DESC } });
+        expect(document.querySelector('.charCounter').classList).toContain("warning");
+    });
+
+    test(`counter almost full appears after ${MAX_DESCRIPTION_LENGTH-7} characters typed`, () => {
```

Comment:
Could this message be a bit clearer? Also can't see this one and the one below in the example logs on the comment.

Reply:
Something like ```(counter almost full) warning appears .... ```

Reply:
Updated PR description with latest layout